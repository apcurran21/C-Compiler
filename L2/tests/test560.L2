(@main
  (@main
    0
    :call_label0
    goto :call_label5
    :call_label1
    rdi <- 3
    call print 1
    return
    :call_label2
    %var2print <- %var2
    %var2print <<= 1
    %var2print <- %var2print
    %var2print += 1
    rdi <- %var2print
    call print 1
    return
    :call_label3
    %var2 <- %var2
    %var2 *= %var1
    %var1 <- %var1
    %var1 -= 1
    %c2 <- 0 < %var1
    
    cjump %c2 = 1 :call_label4
    goto :call_label2
    :call_label4
    %c1 <- %var1 = 0
    
    cjump %c1 = 1 :call_label1
    goto :call_label3
    :call_label5
    %var1 <- 5
    %var1print <- %var1
    %var1print <<= 1
    %var1print <- %var1print
    %var1print += 1
    rdi <- %var1print
    call print 1
    %var2 <- 1
    %var3 <- 0
    goto :call_label4
  )
)
