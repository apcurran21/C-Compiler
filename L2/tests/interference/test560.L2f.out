%arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %arr %arr_start %init %len %ptr r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
%arg2 %arg1 %arg3 %arg4 %arg5 %arg6 %arr %arr_start %init %len %ptr r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
%arg3 %arg1 %arg2 %arg4 %arg5 %arg6 %arr %arr_start %init %len %ptr r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
%arg4 %arg1 %arg2 %arg3 %arg5 %arg6 %arr %arr_start %init %len %ptr r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
%arg5 %arg1 %arg2 %arg3 %arg4 %arg6 %arr %arr_start %init %len %ptr r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
%arg6 %arg1 %arg2 %arg3 %arg4 %arg5 %arr %arr_start %init %len %ptr r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
%arr %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 r12 r13 r14 r15 rbp rbx
%arr_end %arr_i %arr_i_lte_ip1 %arr_ip1 %arr_start %big_num %enc_arr_i %enc_big_num %i %ip1 %loop_end %not_at_end r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
%arr_i %arr_end %arr_i_lte_ip1 %arr_ip1 %arr_start %i %ip1 %loop_end r12 r13 r14 r15 rax rbp rbx
%arr_i_lte_ip1 %arr_end %arr_i %arr_ip1 %arr_start %i %ip1 %loop_end r12 r13 r14 r15 rax rbp rbx
%arr_ip1 %arr_end %arr_i %arr_i_lte_ip1 %arr_start %i %ip1 %loop_end r12 r13 r14 r15 rax rbp rbx
%arr_start %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %arr_end %arr_i %arr_i_lte_ip1 %arr_ip1 %big_num %enc_arr_i %enc_big_num %i %ip1 %loop_end %not_at_end %ptr r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
%big_num %arr_end %arr_start %loop_end r12 r13 r14 r15 rbp rbx
%enc_arr_i %arr_end %arr_start %i %loop_end r12 r13 r14 r15 rbp rbx
%enc_big_num %arr_end %arr_start %loop_end r12 r13 r14 r15 rbp rbx
%i %arr_end %arr_i %arr_i_lte_ip1 %arr_ip1 %arr_start %enc_arr_i %ip1 %loop_end %not_at_end r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
%init %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %len r12 r13 r14 r15 rbp rbx rdi
%ip1 %arr_end %arr_i %arr_i_lte_ip1 %arr_ip1 %arr_start %i %loop_end %not_at_end r12 r13 r14 r15 rax rbp rbx
%len %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %init r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
%loop_end %arr_end %arr_i %arr_i_lte_ip1 %arr_ip1 %arr_start %big_num %enc_arr_i %enc_big_num %i %ip1 %not_at_end r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
%not_at_end %arr_end %arr_start %i %ip1 %loop_end r12 r13 r14 r15 rax rbp rbx
%ptr %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %arr_start r12 r13 r14 r15 rbp rbx
r10 %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %arr_end %arr_start %i %len %loop_end r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
r11 %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %arr_end %arr_start %i %len %loop_end r10 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
r12 %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %arr %arr_end %arr_i %arr_i_lte_ip1 %arr_ip1 %arr_start %big_num %enc_arr_i %enc_big_num %i %init %ip1 %len %loop_end %not_at_end %ptr r10 r11 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
r13 %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %arr %arr_end %arr_i %arr_i_lte_ip1 %arr_ip1 %arr_start %big_num %enc_arr_i %enc_big_num %i %init %ip1 %len %loop_end %not_at_end %ptr r10 r11 r12 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
r14 %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %arr %arr_end %arr_i %arr_i_lte_ip1 %arr_ip1 %arr_start %big_num %enc_arr_i %enc_big_num %i %init %ip1 %len %loop_end %not_at_end %ptr r10 r11 r12 r13 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
r15 %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %arr %arr_end %arr_i %arr_i_lte_ip1 %arr_ip1 %arr_start %big_num %enc_arr_i %enc_big_num %i %init %ip1 %len %loop_end %not_at_end %ptr r10 r11 r12 r13 r14 r8 r9 rax rbp rbx rcx rdi rdx rsi
r8 %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %arr_end %arr_start %i %len %loop_end r10 r11 r12 r13 r14 r15 r9 rax rbp rbx rcx rdi rdx rsi
r9 %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %arr_end %arr_start %i %len %loop_end r10 r11 r12 r13 r14 r15 r8 rax rbp rbx rcx rdi rdx rsi
rax %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %arr_end %arr_i %arr_i_lte_ip1 %arr_ip1 %arr_start %i %ip1 %len %loop_end %not_at_end r10 r11 r12 r13 r14 r15 r8 r9 rbp rbx rcx rdi rdx rsi
rbp %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %arr %arr_end %arr_i %arr_i_lte_ip1 %arr_ip1 %arr_start %big_num %enc_arr_i %enc_big_num %i %init %ip1 %len %loop_end %not_at_end %ptr r10 r11 r12 r13 r14 r15 r8 r9 rax rbx rcx rdi rdx rsi
rbx %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %arr %arr_end %arr_i %arr_i_lte_ip1 %arr_ip1 %arr_start %big_num %enc_arr_i %enc_big_num %i %init %ip1 %len %loop_end %not_at_end %ptr r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rcx rdi rdx rsi
rcx %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %arr_end %arr_start %i %len %loop_end r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rdi rdx rsi
rdi %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %arr_end %arr_start %i %init %len %loop_end r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdx rsi
rdx %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %arr_end %arr_start %i %len %loop_end r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rsi
rsi %arg1 %arg2 %arg3 %arg4 %arg5 %arg6 %arr_end %arr_start %i %len %loop_end r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx
